### ISSUE TITLE
[a11y-defect-3]: Disagreement legend should be descriptive

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#61370]
## Point of contact

@joshkimux

### Severity level

3, Moderate. Should be fixed in 1-3 sprints post-launch.

## Details

Recently, @eileen-coforma informed me of a design change based on content recommendations where the `h3` for the disagreement page will be written as 
> Decision for [issue] dated [date]

From previously on HLR where it was written as 
> What do you disagree with?

This is great! I love the plain language content in the `h3`, but would reason that it should not be the `legend` to the checkboxes below as it's not describing how users should be checking/unchecking the checkboxes.

For example, if a screen reader user were to tab from the `h3` directly to the checkbox, they'd hear the following:

> Decision for [issue] dated [date], heading level 3
> The service connection, decision for [issue] dated [date]

https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/5b09073b-a786-4225-91b2-9ffe0a46a3cb

This wouldn't be ideal, as they'd miss the important instructions that ask them what they disagree with.

## Reproduction steps

1. Log in to NoD as user 236
2. Turn on VoiceOver on Safari
3. Add one issue on the contestable issues page
4. On the disagreement page, turn on VoiceOver
5. Tab to the first checkbox
6. Confirm the detailed label is not read out, only the h3 of the issue

## Proposed solution or next steps

This can be solved in _many_ different ways including, but not limited to:

### Separating the `h3` from the `legend` element, then inserting the "Tell us what you disagree with..." content within the `legend` element. 

This would address the problem, but may be technically difficult (I defer to our engineers to estimate difficulty). In the demo below, I've also added a unique id to the `h3` so I can use `aria-describedby` to uniquely describe each of the checkboxes. I'm not a big fan of this as it adds a bunch of aria to the page.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/26e2dcb4-ffe3-4589-ba4e-cc93e79d2404

### Using the `h3` to contain the full question

This runs counter to the original content suggestion and results in a bloated `h3`. However, it may provide an easier fix to the accessibility issue.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/fcd37985-a253-4637-a51d-02ac1b270438

### A note on `aria-describedby`
Unfortunately, in testing-- I haven't been able to get the `aria-describedby` to work with describing a `legend`. It can, however, work when describing specific checkbox inputs. This feels like a last resort, and we should avoid relying on it when possible for fieldsets like this.

## Acceptance criteria
- [ ] When tabbing to a checkbox, screen reader users hear a unique and descriptive fieldset legend 

## References, articles, or WCAG support

[Use Legend and Fieldset](https://adrianroselli.com/2022/07/use-legend-and-fieldset.html#accDesc)
> You could use aria-describedby to assign the group label to each item within the group as its accessible description. This is not its name, so it can be missed, but it [will at least be exposed](https://adrianroselli.com/2022/04/accessible-description-exposure.html).

> When used with a group role, it will not give quick insight to a user when they have entered a new group, so they will need to listen the full announcement for each control if unfamiliar with the page.

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [X] Focus
- [X] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by eileen-coforma** [comment]
For HLR our issue headers are currently labeled as 'Decision for [issue] at [date]'. 

I think it needs to be changed to ' Disagreement for [issue] at [date]'. I believe that provides more context, despite what the content team initially suggested. I'll run the change by Julie, and confirm after approval.

**Comment by eileen-coforma** [comment]
@mottie New H3 title:
[design disagreement](https://www.sketch.com/s/0424c51d-7818-43ea-8821-6de91eb7ab27/a/GoJ1GpD)
[design review and submit](https://www.sketch.com/s/0424c51d-7818-43ea-8821-6de91eb7ab27/a/GoJ1GpD)
[CSoT](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/decision-reviews/Notice-of-Disagreement/design/sourceofcontenttruth.md#request-a-board-appeal-12)

**Comment by Mottie** [comment]
Work merged in and ready for validation in staging

**Comment by joshkimux** [comment]
Thanks @Mottie ❤️ 

This might need a couple more minor tweaks to get it working. The main issues to be fixed now is that:
- the `h3` is being repeated as both the `legend` and `aria-describedby` to the checkboxes
- the label "Tell us what you disagree..." is not being announced due to `aria-describedby` being spotty when on the `legend` element

<img width="1540" alt="Screenshot 2023-08-03 at 10 11 09 AM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/f6dcbb80-a228-45ae-bf4c-70ca252d3c88">

We can explore the following fixes:

### Option A

1. Separate the h3 from the legend element
2. Insert the "Tell us what you disagree with..." content within the legend element.
3. Remove the `aria-describedby` attribute from the `legend` element as [it does not reliably announce when attached to the `legend` element.](https://blog.tenon.io/accessible-validation-of-checkbox-and-radiobutton-groups/)

### Option B

Would entail editing the `h3` to contain the full question, but would require a content change which we might not have approval for.

Let me know if option A is technically possible, otherwise, I'm happy to bring this up to @eileen-coforma to discuss more!

**Comment by joshkimux** [comment]
## Update 8/4/23

@Mottie informed me that Option A may require significant technical fixes to accomplish, so I asked him to hold off on any fixes until I could think of alternatives. Opting for Option A would result in the input being placed outside of the `fieldset` due to form system constraints.

A way around this that could preserve the existing DOM order would be to:
- keep the `h3` as the `legend`
- set a unique `id` to the paragraph "Tell us what you disagree with..."
- replace the aria-describedby attribute on the checkboxes and input with the unique `id` on the paragraph "Tell us what you disagree with..."
- and remove the aria-describedby attribute from the legend element as [it does not reliably announce when attached to the legend element.](https://blog.tenon.io/accessible-validation-of-checkbox-and-radiobutton-groups/)

This will result in an announcement like this:

> "The service connection, checked checkbox 1 of 3, disagreement with asdf decision on [date]... tell us what you disagree with. You can choose more than one."

The `legend` is announced first, which is the `h3` followed by a brief pause and the question. This reverses the ideal order (the question should come first followed by the description), but still meets the needs of this issue ticket.

@Mottie can confirm if this is less technically intensive to implement next Monday 🙏

**Comment by Mottie** [comment]
@joshkimux Please validate that these changes that are now in place are working as expected.

**Comment by joshkimux** [comment]
https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/e5ecf42a-101a-430e-9908-357054ab58d7

Validated! Thank you so much @Mottie 💙
